wintaser/GUI/Menu.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * (c) 2015- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#include <Windows.h>
+
+#include <cwchar>
+#include <string>
+#include <vector>
+
+#include "Menu.h"
+
+/*
+ * Reference MENUEX_TEMPLATE_HEADER structure
+ * struct MENUEX_TEMPLATE_HEADER
+ * {
+ *     WORD wVersion;
+ *     WORD wOffset;
+ *     DWORD dwHelpId;
+ * };
+ *
+ * Reference MENUEX_TEMPLATE_ITEM structure (PopUp menu)
+ * struct MENUEX_TEMPLATE_ITEM
+ * {
+ *     DWORD dwType;
+ *     DWORD dwState;
+ *     DWORD menuId;
+ *     WORD bResInfo;
+ *     WCHAR szText[];  <-- Variable length NULL terminated WCHAR array
+ *     DWORD dwHelpId;  <-- Not included in the struct when a regular menu item is added
+ * };
+ *
+ * Reference MENUEX_TEMPLATE_ITEM structure (Regular menu-item)
+ * struct MENUEX_TEMPLATE_ITEM
+ * {
+ *     DWORD dwType;
+ *     DWORD dwState;
+ *     DWORD menuId;
+ *     WORD bResInfo;
+ *     WCHAR szText[];  <-- Variable length NULL terminated WCHAR array
+ * };
+ */
+
+namespace {
+    static const DWORD IDC_STATIC = -1;
+    static const DWORD MENUEX_TEMPLATE_HEADER_SIZE = 8;
+    static const DWORD MENUEX_TEMPLATE_ITEM_BASE_SIZE = (sizeof(DWORD) * 3) +
+                                                        (sizeof(WORD) * 1) +
+                                                        (sizeof(WCHAR) * 1);
+};
+
+Menu::Menu()
+{
+    DWORD iterator = 0;
+    menu.resize(MENUEX_TEMPLATE_HEADER_SIZE);
+    *reinterpret_cast<WORD*>(&(menu[iterator])) = 0x01;
+    iterator += sizeof(WORD);
+    *reinterpret_cast<WORD*>(&(menu[iterator])) = 0x04;
+
+    loaded_menu = nullptr;
+}
+
+Menu::~Menu()
+{
+    if (loaded_menu != nullptr)
+    {
+        DestroyMenu(loaded_menu);
+    }
+}
+
+void Menu::AddMenuCategory(std::wstring name, DWORD id, bool enabled, bool last)
+{
+    DWORD state = ((enabled) ? MFS_ENABLED : MFS_DISABLED);

```

Comment:
Why the parentheses around "enabled"? Same goes for similar cases in rest of file.